scipy/sparse/linalg/_dsolve/tests/test_linsolve.py

```diff
@@ -774,29 +774,33 @@ def test_input_types(self):
     @pytest.mark.timeout(120)  # prerelease_deps_coverage_64bit_blas job
     @sup_sparse_efficiency
     def test_random(self):
-        def random_triangle_matrix(n, lower=True):
-            A = scipy.sparse.random(n, n, density=0.1, format='coo')
+        def random_triangle_matrix(n, lower=True, format="csr"):
+            A = scipy.sparse.random(n, n, density=0.1, format='lil')
```

Comment:
This code can be accelerated using the new NumPy generators. As this gets executed several times, it is worth it in my opinion.

```python
import scipy
import numpy as np

rng = np.random.default_rng()

rvs = rng.random

n = 10**3

%timeit A = scipy.sparse.random(n, n, density=0.1, format='lil')  # 29.4 ms
%timeit A = scipy.sparse.random(n, n, density=0.1, format='lil', random_state=rng, data_rvs=rvs)  # 16.4 ms
```

Reply:
Also at least in `scipy.stats`, fixed random seeds are often preferred over new random data for every test run. Do we have general guidelines for this in scipy?

Reply:
> This code can be accelerated using the new NumPy generators. 

Thanks, changed in 8aceb5ac9a26b77e966330560d1b2bee1f538774.

> Also at least in `scipy.stats`, fixed random seeds are often preferred over new random data for every test run. 

Also in [test_linsolve.py](https://github.com/scipy/scipy/blob/bf819ba58a6ccd51cc6d6dacf88ab18a80355148/scipy/sparse/linalg/_dsolve/tests/test_linsolve.py), constant seeds are used in several places via `random.seed` or `random.RandomState`. I therefore chose to also fix the seed here, see 37ee66609dbbcc3852d7a1b1282e27cd187745d8. The particular random seed used here has been selected at random.